z3rro/src/prover.rs

```diff
@@ -13,6 +25,23 @@ use crate::{
     util::{set_solver_timeout, ReasonUnknown},
 };
 
+
+#[derive(Debug, Error)]
+pub enum ProverCommandError {
+    #[error("Environment variable error: {0}")]
+    EnvVarError(#[from] env::VarError),
+    #[error("Process execution failed: {0}")]
+    ProcessError(#[from] io::Error),
+    #[error("Parse error")]
+    ParseError,
+}
+
+#[derive(Debug)]
+pub enum SolverType {
```

Comment:
Let's introduce another SolverType here: `SMTLIB(Command)`, which takes a `Command` object and uses it as a backend, just like we currently do for SWINE. Except that for `SMTLIB`, we don't remove the `exp` declaration.